File: Algorithms/Sort/Bubble sort/Bubble sort.cpp
#include <iostream>

using namespace std;

void bubble_sort(float list[], int n){
	for (int i = 0; i < n - 1; i++){
		for (int j = 0; j < n - i - 1; j++){
			if (list[j] < list[j + 1]){
				float temp = list[j];
				list[j] = list[j + 1];
				list[j + 1] = temp;
			}
		}
	}
}

int main(){
	
	int n;
	cout << "n = "; cin >> n;
	float list[n];
	
	cout << "List to sort: ";
	for (int i = 0; i < n; i++){
		cin >> list[i];
	}	
	
	bubble_sort(list, n);
	cout << "List after sorting:";
	for (int i = 0; i < n; i++){
		cout << " " << list[i];
	}		
	
}

File: Algorithms/Sort/Bubble sort/Bubble sort.py
def bubble_sort(list_to_sort):
    for i in range(0, len(list_to_sort) - 1):
        for j in range(0, len(list_to_sort) - i - 1):
            if list_to_sort[j] > list_to_sort[j + 1]:
                temp = list_to_sort[j]
                list_to_sort[j] = list_to_sort[j + 1]
                list_to_sort[j + 1] = temp
                

list0 = input("List to sort: ")
list1 = list(map(float, list0.strip().split()))       
                
bubble_sort(list1)
list1 = ' '.join(map(lambda x: str(int(x)) if x.is_integer() else str(x), list1))
print("List after sorting: ", list1)                